tests/Doctrine/Tests/Common/Cache/CacheTest.php

```diff
@@ -29,6 +29,21 @@ public function testBasicCrudOperations($value)
         $this->assertFalse($cache->contains('key'));
     }
 
+    public function testCacheKeyIsCaseSensitive()
+    {
+        $cache = $this->_getCacheDriver();
+
+        $this->assertTrue($cache->save('key', 'value'));
+        $this->assertTrue($cache->contains('key'));
+        $this->assertSame('value', $cache->fetch('key'));
+
+        $this->assertFalse($cache->contains('KEY'));
+        $this->assertFalse($cache->fetch('KEY'));
+
+        $cache->delete('KEY');
+        $this->assertTrue($cache->contains('key', 'Deleting cache item with different case must not affect other cache item'));
```

Comment:
i know but that cannot be avoided. there probably is key with different case that has the same first 2 chars in the hash. I I don't know a way to find this collision. So this is more a theoretical test that will be even more relavant in case we make the balancing hash logic configurable/optional.